test/src/client_utils.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:http2/transport.dart';
+import 'package:test/test.dart';
+import 'package:mockito/mockito.dart';
+
+import 'package:grpc/grpc.dart';
+
+class MockConnection extends Mock implements ClientTransportConnection {}
+
+class MockStream extends Mock implements ClientTransportStream {}
+
+class MockChannel extends Mock implements ClientChannel {}
+
+typedef ServerMessageHandler = void Function(StreamMessage message);
+
+class TestClient {
+  final ClientChannel _channel;
+
+  static final _$unary = new ClientMethod<int, int>(
+      '/Test/Unary',
+      (int value) => new List.filled(value, 0),
```

Comment:
Extract these as `mockEncode` and `mockDecode` functions to use in the client methods and in the request handlers in the tests.